### PR TITLE
fix: issue 7166 RELEASE

### PR DESCRIPTION
**Pull request by nirgur**
## Related Issues

Fixes https://github.com/descope/etc/issues/7166

Verified the scenario in the issue, 
In this case there is no response because there is no network so no error event will be dispatched,
Also added a default error object in case we did not get one from the server with the default values

### CODE REVIEW

**Review by dorsha — DISMISSED** [review]
Cool
1. Build fails
2. What about increasing the timeout and adding retries? Can be done in another pr

File: packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts [hunk 1]
```diff
@@ -307,6 +307,14 @@ class BaseDescopeWc extends HTMLElement {
         try {
           const resp = await origFn(...args);
           return resp;
+        } catch (e) {
+          // return a generic error object in case of an error
+          return {
+            error: {
+              errorCode: 'E151000',
```
Comment (orius123) [inline packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts:314]: Use a different error code, so we can tell it is coming from here

Comment (nirgur) [reply packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts:314]: Which one?

Comment (asafshen) [reply packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts:314]: good point Eliya, `E` is used by Descope service,

in go-sdk Descope uses `G...` for generated errors (not sure g stands for generated or go btw)

image


maybe `G` or dedicated prefix for this (`J` for JS?)

Comment (nirgur) [reply packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts:314]: This is what @dorsha suggested in the issue
https://github.com/descope/etc/issues/7166#issuecomment-2249727456

Comment (dorsha) [reply packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts:314]: We can use J

Comment (nirgur) [reply packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts:314]: So `J151000` in both places?

Comment (orius123) [reply packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts:314]: `J151000` here and `J151001` in `packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts` if we don't have any error at all

Comment (nirgur) [reply packages/sdks/web-component/src/lib/descope-wc/BaseDescopeWc.ts]: Done

**Review by orius123 — APPROVED** [review]

**Comment by dorsha** [comment]
@nirgur build fails

**Comment by orius123** [comment]
Don't we wamt to also release it?

**Comment by nirgur** [comment]
> Cool
> 
> 1. Build fails
> 2. What about increasing the timeout and adding retries? Can be done in another pr

Let's talk about it first and decide how to handle

**Comment by nirgur** [comment]
> Don't we wamt to also release it?

Released

**Pull request merged**